### DISCUSSION TITLE
Does remix-flat-routes also work for Remix run version 1.x (e.g.: 1.9.x)?

### DISCUSSION THREAD

**Discussion by mtin79** [kiliman/remix-flat-routes#95] [Q&A] [answered]
Just a simple question before i try to get this working in production with remix run 1.9.x.
Does remix-flat-routes also work for Remix run version 1.x (e.g.: 1.9.x)?
How would i need to set up the configuration? Same as for v2?
Thanks a lot.

**Comment by kiliman** [comment, marked as answer]
Yes, `remix-flat-routes` works with all versions of Remix from v1.5+.

The simplest way to configure it is to just copy the configuration from the README.

Just make sure to set the `ignoredRouteFiles` to ignore _ALL_ files, so Remix doesn't try to process them before `remix-flat-routes`.

```ts
const { flatRoutes } = require('remix-flat-routes')

/**
 * @type {import("@remix-run/dev").AppConfig}
 */
module.exports = {
  // ignore all files in routes folder to prevent
  // default remix convention from picking up routes
  ignoredRouteFiles: ['**/*'],
  routes: async defineRoutes => {
    return flatRoutes('routes', defineRoutes)
  },
}
```
If you're using ESM, then just convert the syntax to:
```ts
import { flatRoutes } from 'remix-flat-routes'

/**
 * @type {import("@remix-run/dev").AppConfig}
 */
export default {
  // ignore all files in routes folder to prevent
  // default remix convention from picking up routes
  ignoredRouteFiles: ['**/*'],
  routes: async defineRoutes => {
    return flatRoutes('routes', defineRoutes)
  },
}
```

**Reply by mtin79** [reply]
Wonderful, thank you!

**Reply by dev-1tm-solutions** [reply]
Hi @kiliman ,
just one more thing to clarify. In the README it says: "remix-flat-routes will always maintain compatibility with the default Remix convention. This package is simply a superset/extension of the core convention."
If i have a remix run v1, doest remix-flat-routes also support the remix routes convention of version 1?

**Reply by kiliman** [reply]
Sorry, I should clarify that `remix-flat-routes` is compatible with Remix v2+. 

If you want `v1` routes, then you can use the Remix compat package: `@remix-run/v1-route-convention`

https://github.com/remix-run/v1-compat-utils/tree/main/packages/v1-route-convention
